### PR TITLE
Initial recipe submission page

### PR DESCRIPTION
**Pull request by heath3n**
Still needs error handling, user feedback, and some validation but everything should be working with Firebase if you turn off security rules.

### CODE REVIEW

File: src/components/RecipeSubmit.vue [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        photos: [],
+        tags: [],
+        recipeSteps: [
+          {}
+        ]
+      };
+    },
+    mounted: function() {
+      // Load cookware dropdown options
+      var cookware = [];
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Use 'const' or 'let' in place of 'var'.
Use 'const' when the variable is never reassigned.
Use 'let' if you plan to reassign that variable.

File: src/components/RecipeSubmit.vue [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      db.ref('cookware/').orderByChild('description').on('child_added', function(c) {
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Define database refs as const variables before the export block

File: src/components/RecipeSubmit.vue [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        if (!files.length) {
+          return;
+        }
+
+        var storageRef = Firebase.storage().ref(); // TODO: How do I put this in database.js like we did with db?
+
+        var fileList = [];
+        var photoList = [];
+
+        for (var i = 0; i < files.length; i++) {
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Prefer ES6 'Array.prototype.forEach' when you want to iterate over items in an array. Ex:
files.forEach((file) => {
    // do stuff
});

File: src/components/RecipeSubmit.vue [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+          fileList.push({src: window.URL.createObjectURL(files[i])});
+
+          var uploadTask = storageRef.child('images/recipes/' + uuidV4()).put(files[i]);
+          uploadTask.on('state_changed', function(snapshot) {
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Prefer ES6 arrow functions when value of 'this' does not matter. Ex:
uploadTask.on('state_changed', (snapshot) => {

File: src/components/RecipeSubmit.vue [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+            // TODO: Show upload progress?
+          }, function(error) {
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Prefer arrow function

File: src/components/RecipeSubmit.vue [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    </div>
+    </v-app>
+</template>
+
+<script>
+  import Firebase from 'firebase';
+  import db from '../database';
+  import router from 'vue-router';
+  import uuidV4 from 'uuid/v4';
+  import VSelect from "vuetify/es5/components/VSelect/VSelect";
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Replace all double quotes literals with single quotes literals, unless you're specifically writing a string that contains a single quote.

File: src/components/RecipeSubmit.vue [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+        this.$nextTick(() => (this.showCarousel = true));
+      },
+      // TODO: Handle undefineds since Firebase will reject the whole request if they're present.
+      submitRecipe() {
+        db.ref().child('recipes').push().set({
+          submitter: Firebase.auth().currentUser.uid,
+          name: this.recipeName,
+          description: this.recipeDescription,
+          ingredients: this.ingredients.filter(x => x.trim() != ''),
+          prepTime: (parseInt(this.prepTimeHr) * 60 + parseInt(this.prepTimeMin)),
```
Comment (steplica) [inline src/components/RecipeSubmit.vue]: Explicitly parenthesize arithmetic to make it easier to read. Ex:
prepTime: (parseInt(this.prepTimeHr) * 60) + parseInt(this.prepTimeMin),

File: src/components/RecipeSubmit.vue [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+        </div>
+    </div>
+    </v-app>
+</template>
+
+<script>
+  import Firebase from 'firebase';
+  import db from '../database';
+  import router from 'vue-router';
+  import uuidV4 from 'uuid/v4';
```
Comment (steplica) [inline src/components/RecipeSubmit.vue:135]: Are you using this to generate unique identifier strings to store recipes in the DB? If so, firebase actually has native support for this.

Comment (heath3n) [reply src/components/RecipeSubmit.vue:135]: I know it autogenerates those for database entries but it requires a filename for storing images and according to their PM, [they didn't want to assume how devs wanted files to be named](https://stackoverflow.com/questions/37444685/store-files-with-unique-random-names). At the moment I'm just dumping them all into `images/recipes/` but that will probably change.

**Pull request merged**